### PR TITLE
fix(ui5-application-writer): use correct framework url #290

### PR DESCRIPTION
**Pull request by korotkovao**
https://github.com/SAP/open-ux-tools/issues/290

Fix `frameworkUrl` logic.
Add unit tests.

### CODE REVIEW

**Review by devinea — COMMENTED** [review]
Thanks @korotkovao.
Changes look good so passed in values are not  over written by defaults. 
I linked https://github.com/SAP/open-ux-tools/issues/290 in the description.
Can you add a changeset per https://github.com/SAP/open-ux-tools#create-changesets-for-feature-or-bug-fix-branches ?

**Comment by korotkovao** [comment]
Thanks @devinea

**Review by devinea — APPROVED** [review]
Thanks @korotkovao 
We can ignore the sonar coverage as there seems to be some issue there that can be looked into separately. Coverage in the build is good.
https://github.com/SAP/open-ux-tools/runs/5021309855?check_suite_focus=true
Screenshot 2022-02-01 at 14 56 30

File: packages/ui5-application-writer/src/data/defaults.ts [hunk 1]
```diff
@@ -61,29 +61,35 @@ export enum UI5_DEFAULT {
...
+    merged.typesVersion = ui5.typesVersion ?? typesVersion;
+    merged.ui5Theme = ui5.ui5Theme ?? 'sap_fiori_3';
+    merged.ui5Libs = ui5.ui5Libs ?? [];
 
     return Object.assign({}, ui5, merged) as UI5;
 }
 
 // if a specific local version is provided, use it, otherwise, sync with version but keep minimum versions in mind
+/**
+ * @param root0 input object
```
Comment (IainSAP) [inline packages/ui5-application-writer/src/data/defaults.ts:87]: What is `root0`?

**Review by IainSAP — APPROVED** [review]

**Review by devinea — APPROVED** [review]

**Comment by korotkovao** [comment]
Thanks, @devinea and @IainSAP!

**Pull request merged**